core/arch/arm/mm/sp_mem.c

```diff
@@ -17,6 +16,138 @@ static bitstr_t bit_decl(share_bits, NUM_SHARES);
 /* mem_shares stores all active FF-A shares. */
 SLIST_HEAD(sp_mem_head, sp_mem);
 static struct sp_mem_head mem_shares = SLIST_HEAD_INITIALIZER(sp_mem_head);
+const struct mobj_ops mobj_sp_ops;
+
+struct mobj_sp {
+	struct mobj mobj;
+	paddr_t *pages;
+};
+
+static struct mobj_sp *to_mobj_sp(struct mobj *mobj)
+{
+	assert(mobj->ops == &mobj_sp_ops);
+	return container_of(mobj, struct mobj_sp, mobj);
+}
+
+struct mobj *sp_mem_new_mobj(uint64_t pages)
+{
+	struct mobj_sp *m = NULL;
+
+	m = malloc(sizeof(struct mobj_sp));
+	if (!m)
+		return NULL;
+
+	m->mobj.ops = &mobj_sp_ops;
+	m->mobj.size = pages * SMALL_PAGE_SIZE;
+	m->mobj.phys_granule = SMALL_PAGE_SIZE;
+	m->pages = malloc(pages * sizeof(pages));
```

Comment:
We should probably make sure to clear these since garbage here can be catastrophic and hard to track down.

Reply:
Are you talking about the pages? These get cleaned-up in the `inactivate()` function.

Reply:
Please use `calloc()` or `memset()` to make sure that eventual gaps in the ranges processed in `spmc_sp_add_nw_region()` does't cause unchecked values to be used as physical addressed.

Reply:
Done